dayana-Day-ana/homework_7/zippopotamus/script.js

```diff
@@ -0,0 +1,50 @@
+/*
+    Use the zippopotam.us service to look up and display the city and state for a zip code that users enter in the Postal Code box on the form.
+    zippopotam.us does not require an API key. See the website zippopotam.us for API documentation for the U.S.
+    Start by logging the city and state to the console. If you wish, you can enhance your code using DOM manipulation to populate the City and State input boxes with the data.
+    Convert your event handler code to an IIFE and verify that your app still works the same way.
+    Convert all of your JS code to a single IIFE and verify that the app still works the same way. 
+    Bonus: 
+      Instead of responding to a button click, add code to listen for the keyup event in the Postal Code box after a keyboard key is pressed and released, and check if the Postal Code box contains sufficient characters for a zip code before performing a search automatically.
+*/
+
+"use strict";
+
+$("#zip").keyup(function(){
+
+	//call
+	var el = $(this);
+	if ((el.val().length == 5)) {
+    	//When the value is equal to 5 call look up function
+    	lookUp();
```

Comment:
Excellent use of a callback and abstracting away `lookUp` code to be activated on `.keyup`!

Including explicit details in comments helps a lot when other people are reading your code. Here the context of zipcode length it's clear that the value is zipcode, but in a larger scale application that might not be as apparent. 

One tiny thing: `==` checks to see if one value is the same as another, whereas `===` checks to see the value is the same, _and_ that the type is the same. JavaScript was built initially with `type coercion` as a feature, which allows silly things like returning true when comparing a string and number. This is an absolutely ridiculous feature because fuzzes your data types, which isn't best practice.

Allow yourself to be a better developer - `'use strict` and `===` your code. 

```
> 5 == "5"
true
> 5 === "5"
false
```